code/camera/cameradisplay.cpp

```diff
@@ -117,6 +127,28 @@ void CameraDisplay::update_zoom(int value) {
     Q_EMIT zoom_changed(zoom_factor);
 }
 
+void CameraDisplay::gridSelectChanged(int weight_index) {
+        weightSelected = m_ui->weight_list->currentText();
+     //   m_ui->grid_display->selectRobotPosition(weightSelected);
+    Q_EMIT select_position(weightSelected);
+    #ifndef NDEBUG
+        qDebug() << "Grid selector option changed to " << QString(weightSelected);
+    #endif
+}
+
+void CameraDisplay::weightingChanged(int weighting) {
+    //get number from weight-entering box
+        //    weighting = getWeighting();
+#ifndef NDEBUG
+        qDebug() << "Weight Changed";
+#endif
+}
+
+int CameraDisplay::getWeighting() {
+        weighting = m_ui->weight_selector->value();
```

Comment:
Wouldn't it be better to store the changed weight in `weightingChanged` and then just return that here? It might help avoid issues later down the line if another thread calls `getWeighting` and tries to access `weight_selector`

also snake case naming